client/state/themes/test/actions.js

```diff
@@ -55,20 +53,39 @@ import {
 	receiveRecommendedThemes,
 } from '../actions';
 
-describe( 'actions', () => {
-	const spy = sinon.spy();
+expect.extend( {
+	toMatchFunction( received, fn ) {
+		if ( received.toString() === fn.toString() ) {
+			return {
+				message: () => 'Expected functions to match.',
+				pass: true,
+			};
+		}
 
-	function isEqualFunction( f1, f2 ) {
-		// TODO: Also compare params!
-		return f1.toString() === f2.toString();
-	}
+		return {
+			message: () => 'Expected functions to not match.',
```

Comment:
I'm a bit confused by this one -- so we _expect_ the functions to not match, but the test does not pass if the functions are not equal?

Reply:
Yeah, the jest [custom matchers API](https://jestjs.io/docs/expect#custom-matchers-api) isn't very straightforward:

> `pass` indicates whether there was a match or not, and `message` provides a function with no arguments that returns an error message in case of failure.